### ISSUE TITLE
Restocking in hand breaks inventory cell during connection issues. Combining this with shulker chests breaks the inventory entirely and requires relogin.

### ISSUE THREAD

**Issue by Megaddd** [Inventory-Tweaks/inventory-tweaks#480]
**Minecraft Version:** 112
**Inventory Tweaks Version:** 1.63+dev.94.3cdd29e

If client-server communication isn't smooth when an item is restocked as the hand runs out, the item appears as restocked for a moment, but then turns into a blank spot in the restocked slot, even though the item is actually there server-side, and clicking on said blank slot synchronizes the inventory and the item appears on the cursor as expected. 

In some cases, such as restocking shulker chests this can cause a weird issue where you are unable to put anything into that hot-bar slot, and anything you try to place out of it will instead be replaced by a shulker chest.

**Comment by Kobata** [comment]
Minecraft's inventory sync is a bit weird like that sometimes, did this happen on a vanilla server or one with forge + inventory tweaks installed?

I can't do much about vanilla, unfortunately, but with the server-side module it *should* be delaying syncs until after the server has received all the item movements.

**Comment by Megaddd** [comment]
Client mods only, forge+litemod. Let me get you a gif, gimmie 15min.

**Comment by Megaddd** [comment]
Here's what I'm talking about in action: 
Restocked items get stuck on inventory cursor - https://gfycat.com/ReadyPleasantAdolescent
Rapidly restocked items don't show in the world - https://gfycat.com/UnsungKeyWaterlogged

**Comment by Kobata** [comment]
Like I said, if you're talking a vanilla server, there's not very much that can be done -- it's just that minecraft is kinda horrible at syncing the inventory (from the server) sometimes when there's a load of rapid changes -- and some servers make it worse by having plugins that will revert your inventory when they find things happening 'too fast'

**Comment by Megaddd** [comment]
Yeah the server is very bare-bones vanilla, no plugins of any kind.
